src/components/Api.js

```diff
@@ -0,0 +1,160 @@
+export default class Api {
+  constructor(data) {
+    this.baseUrl = data.baseUrl;
+    this.headers = data.headers;
+  }
+
+  // Method to get initial cards
+  getInitialCards() {
+    return fetch(`${this.baseUrl}/cards`, {
+      headers: this.headers,
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
+        console.error("Failed to fetch initial cards:", err);
+      });
+  }
+
+  // Method to get user info
+  getUserInfo() {
+    return fetch(`${this.baseUrl}/users/me`, {
+      headers: this.headers,
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
+        console.error("Failed to fetch user info:", err);
+      });
+  }
+
+  // Method to update user profile
+  updateUserInfo(name, about) {
+    return fetch(`${this.baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: {
+        ...this.headers,
+        "Content-Type": "application/json",
```

Comment:
According to the task you need to pass full `headers` into the constructor of `Api`. See here https://snipboard.io/pRdaAZ.jpg . And in the code you need to use only: `headers: this._headers` (not to duplicate the code)